newton/examples/sensors/example_sensor_tiled_camera.py

```diff
@@ -38,175 +40,224 @@
+                )
+            if rng.random() < 0.5:
+                builder.add_shape_box(
+                    builder.add_body(xform=wp.transform(p=wp.vec3(-2.0, 2.0, 0.5), q=wp.quat_identity())),
+                    hx=0.5,
+                    hy=0.35,
+                    hz=0.5,
+                )
+            if rng.random() < 0.5:
+                builder.add_shape_mesh(
+                    builder.add_body(xform=wp.transform(p=wp.vec3(0.0, 4.0, 0.0), q=wp.quat(0.5, 0.5, 0.5, 0.5))),
+                    mesh=bunny_mesh,
+                )
+            builder.add_builder(robot_builder)
+            builder.end_world()
 
-        self.mesh_pos = wp.vec3(0.0, 4.0, 0.0)
-        body_mesh = builder.add_body(xform=wp.transform(p=self.mesh_pos, q=wp.quat(0.5, 0.5, 0.5, 0.5)), key="mesh")
-        builder.add_shape_mesh(body_mesh, mesh=demo_mesh)
+        builder.add_ground_plane()
 
-        # finalize model
         self.model = builder.finalize()
         self.state = self.model.state()
 
         self.viewer.set_model(self.model)
 
+        sensor_render_width = int(self.viewer.ui.io.display_size[0] // self.num_worlds_per_row)
+        sensor_render_height = int(self.viewer.ui.io.display_size[1] // self.num_worlds_per_col)
+
         # Setup Tiled Camera Sensor
         self.tiled_camera_sensor = TiledCameraSensor(
             model=self.model,
             num_cameras=1,
-            width=640,
-            height=360,
+            width=sensor_render_width,
+            height=sensor_render_height,
             options=TiledCameraSensor.Options(
                 default_light=True, default_light_shadows=True, colors_per_shape=True, checkerboard_texture=True
             ),
         )
+
+        fov = 45.0
         if isinstance(self.viewer, ViewerGL):
-            self.camera_rays = self.tiled_camera_sensor.compute_pinhole_camera_rays(
-                math.radians(self.viewer.camera.fov)
-            )
-        else:
-            self.camera_rays = self.tiled_camera_sensor.compute_pinhole_camera_rays(math.radians(45.0))
+            fov = self.viewer.camera.fov
+
+        self.camera_rays = self.tiled_camera_sensor.compute_pinhole_camera_rays(math.radians(fov))
         self.tiled_camera_sensor_color_image = self.tiled_camera_sensor.create_color_image_output()
         self.tiled_camera_sensor_depth_image = self.tiled_camera_sensor.create_depth_image_output()
-        self.create_textures()
+        self.create_texture()
 
     def step(self):
-        pass
+        wp.launch(
+            animate_franka,
+            self.model.joint_count,
+            [
+                self.time,
+                self.model.joint_type,
+                self.model.joint_dof_dim,
+                self.model.joint_q_start,
+                self.model.joint_qd_start,
+                self.model.joint_q,
+                self.model.joint_limit_lower,
+                self.model.joint_limit_upper,
+            ],
+        )
```

Comment:
For clarity it might be better to write to `state.joint_q` and explicitly list it under the `outputs` argument of the kernel launch. Then use the `state` arrays rather than `model` arrays as source inputs for `eval_fk` below.